### PR TITLE
Searching repos

### PR DESCRIPTION
**Pull request by hahmed**
Let me kickstart my documentation contribution with this PR.

Relates to #948

Let me know what you want me to improve on.

/cc @shiftkey

This PR negates #950

### CODE REVIEW

File: docs/search.md [hunk 1]
```diff
@@ -115,11 +115,79 @@ var request = new SearchIssuesRequest("linux")
...
+    In = new[] { InQualifier.Readme, InQualifier.Description, InQualifier.Name },
+    
+    // how about searching for libraries created before a given date?
+    Created = DateRange.GreaterThan(new DateTime(2015, 1, 1)),
+    
+    // or after a given date?
+    Created = DateRange.LessThan(new DateTime(2015, 1, 1)),
+    
+    // or maybe the last time this repo was updated?
+    Updated = DateRange.GreaterThan(new DateTime(2013, 1, 1)),
```
Comment (shiftkey) [inline docs/search.md]: If we're demonstrating `DateRange` here could we drop this usage and just include `DateRange.Between` here?

**Comment by shiftkey** [comment]
Just one little question. Everything else is :gem:

File: docs/search.md (hunk 1 above)
Comment (hahmed) [inline docs/search.md]: That's cool...

Whilst I was writing this, I was thinking... will you want us to describe the ````daterange```` class as this is a documentation, or for now we will simply expand on the documentation as and when?

**Comment by hahmed** [comment]
@shiftkey That's done now.

File: docs/search.md [hunk 2]
```diff
@@ -115,11 +115,70 @@ var request = new SearchIssuesRequest("linux")
 };
 ```
 
+## Search Repositories
 
+Finding repositories via various criteria.
```
Comment (shiftkey) [inline docs/search.md]: This seems a bit basic - could we jazz this up a bit?

e.g.

> To search for repositories using Octokit you need to create a `SearchRepositoriesRequest` and populate it with the search criteria.

File: docs/search.md [hunk 3]
```diff
@@ -115,11 +115,70 @@ var request = new SearchIssuesRequest("linux")
...
+    // sort by the number of stars
+    SortField = RepoSearchSort.Stars,
+
+    // or by forks?
+    SortField = RepoSearchSort.Forks,
+
+    // how about changing that sort direction?
+    Order = SortDirection.Ascending
+}
+```
```
Comment (shiftkey) [inline docs/search.md]: It'd be nice to have an example below this around how to use this with a `GitHubClient`, if only to see something like:

```csharp
var results = await gitHubClient.Search.SearchRepositories(request);
```

or whatever it's supposed to do...

**Comment by shiftkey** [comment]
Sorry, a couple more things while I'm in here.

File: docs/search.md (hunk 3 above)
Comment (hahmed) [inline docs/search.md]: That's cool, would this be better served at the top? I mean... assuming that we have already told the user about creating a githubclient, we could simply say at the top of this page. 
````
 var results = await gitHubClient.Search.SearchRepositories(request);
````
Then pass in any of the request objects below e.g. ````SearchRepositoriesRequest```` etc. What do you think?

Comment (shiftkey) [inline docs/search.md]: > That's cool, would this be better served at the top?

Yeah! Let's do that!

**Comment by hahmed** [comment]
Let me know how that change works for ya.

File: docs/search.md [hunk 4]
```diff
@@ -115,11 +115,73 @@ var request = new SearchIssuesRequest("linux")
 };
 ```
 
+## Search Repositories
 
+To search for repositories using Octokit you need to create a ```SearchRepositoriesRequest``` and populate it with the search criteria.
```
Comment (shiftkey) [inline docs/search.md]: :lipstick: single ` here should work fine too

File: docs/search.md [hunk 5]
```diff
@@ -115,11 +115,73 @@ var request = new SearchIssuesRequest("linux")
...
 
-## Search Repositories
+```csharp
+// Use your github client to search for repos by passing in a SearchRepositoriesRequest instance
+var result = githubClient.Search.SearchRepo(request);
```
Comment (shiftkey) [inline docs/search.md]: `await` ?

Comment (shiftkey) [inline docs/search.md]: And should this be *after* the `request` ctors defined below?

File: docs/search.md [hunk 6]
```diff
@@ -115,11 +115,73 @@ var request = new SearchIssuesRequest("linux")
 };
 ```
 
+## Search Repositories
 
+To search for repositories using Octokit you need to create a ```SearchRepositoriesRequest``` and populate it with the search criteria.
 
-## Search Repositories
+```csharp
+// Use your github client to search for repos by passing in a SearchRepositoriesRequest instance
```
Comment (shiftkey) [inline docs/search.md]: Let's not worry about the comment here.

**Comment by hahmed** [comment]
Ok done those fixes.

**Pull request merged**